src/modules/routing/RecipientModule.ts

```diff
@@ -0,0 +1,193 @@
+import { KeylistUpdateResponseHandler } from './handlers/KeylistUpdateResponseHandler'
+import { MediationDenyHandler } from './handlers/MediationDenyHandler'
+import { MediationGrantHandler } from './handlers/MediationGrantHandler'
+import { BatchPickupMessage } from './messages/BatchPickupMessage'
+import { MediationState } from './models/MediationState'
+import { RecipientService } from './services/RecipientService'
+
+@scoped(Lifecycle.ContainerScoped)
+export class RecipientModule {
+  private agentConfig: AgentConfig
+  private recipientService: RecipientService
+  private connectionService: ConnectionService
+  private messageSender: MessageSender
+  private eventEmitter: EventEmitter
+  private $stop: Subject<boolean>
+
+  public constructor(
+    dispatcher: Dispatcher,
+    agentConfig: AgentConfig,
+    recipientService: RecipientService,
+    connectionService: ConnectionService,
+    messageSender: MessageSender,
+    eventEmitter: EventEmitter,
+    @inject(InjectionSymbols.$Stop) $stop: Subject<boolean>
+  ) {
+    this.agentConfig = agentConfig
+    this.connectionService = connectionService
+    this.recipientService = recipientService
+    this.messageSender = messageSender
+    this.eventEmitter = eventEmitter
+    this.registerHandlers(dispatcher)
+    this.$stop = $stop
+  }
+
+  public async initialize() {
+    const { defaultMediatorId, clearDefaultMediator } = this.agentConfig
+
+    // Set default mediator by id
+    if (defaultMediatorId) {
+      const mediatorRecord = await this.recipientService.getById(defaultMediatorId)
+      await this.recipientService.setDefaultMediator(mediatorRecord)
+    }
+    // Clear the stored default mediator
+    else if (clearDefaultMediator) {
+      await this.recipientService.clearDefaultMediator()
+    }
+
+    // Poll for messages from mediator
+    const defaultMediator = await this.findDefaultMediator()
+    if (defaultMediator) {
+      await this.initiateMessagePickup(defaultMediator)
+    }
+  }
+
+  public async initiateMessagePickup(mediator: MediationRecord) {
+    const { mediatorPickupStrategy, mediatorPollingInterval } = this.agentConfig
+
+    const mediatorConnection = await this.connectionService.getById(mediator.connectionId)
+
+    // Explicit means polling every X seconds with batch message
+    if (mediatorPickupStrategy === MediatorPickupStrategy.Explicit) {
+      this.agentConfig.logger.info(`Starting explicit (batch) pickup of messages from mediator '${mediator.id}'`)
+      const subscription = interval(mediatorPollingInterval)
+        .pipe(takeUntil(this.$stop))
+        .subscribe(async () => {
+          await this.pickupMessages(mediatorConnection)
+        })
+
+      return subscription
+    }
+
+    // Implicit means sending ping once and keeping connection open. This requires a long-lived transport
+    // such as WebSockets to work
+    else if (mediatorPickupStrategy === MediatorPickupStrategy.Implicit) {
```

Comment:
I think this will change as the transports are updated to support multiple protocols. I am resolving this conversation in favor of creating an issue as needed. Please tag me in any issues or further conversations.